src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -47,8 +47,31 @@ def inner_cli_command(prefix, cmd_args="", desc=""):
 
 class OrchestratorCli(orchestrator.OrchestratorClientMixin, MgrModule):
     MODULE_OPTIONS = [
-        {'name': 'orchestrator'}
+        {
+            'name': 'orchestrator',
+            'default': None,
+            'desc': 'Orchestrator backend',
+            'enum_allowed': ['ssh', 'rook', 'ansible', 'deepsea'],
+            'runtime': True,
+        },
     ]
+    NATIVE_OPTIONS = []
+
+    def config_notify(self):
+        """
+        This method is called whenever one of our config options is changed.
+        """
+        for opt in self.MODULE_OPTIONS:
```

Comment:
I guess every module would benefit from this code, right?

Reply:
Probably, but (1) we need to do a big conversion if we move it into mgr_module, and (2) some modules still will need to implement this hook so that they wake up when such a change happens.